js/main.js

```diff
@@ -746,6 +748,25 @@ class App extends React.Component {
     this.setState({cols: cols, width: width}, () => {this.relayout()});
   }
 
+  generateWindowHash = (windowId) => {
+    let windowContent = this.state.panes[windowId];
+
+    //Convert JSON data to string with a space of 2. This detail is important.
```

Comment:
Please note why this detail is important (keeping the js hash consistent with the python hash if I'm guessing correctly).

Reply:
Yup. This is to ensure that the JSON is converted to identical strings in both the implementations.